apps/condo/domains/meter/schema/RegisterMetersReadingsService.js

```diff
@@ -298,13 +317,36 @@ const RegisterMetersReadingsService = new GQLCustomSchema('RegisterMetersReading
 
                 const resultRows = []
 
+                const meterNumbers = uniq(readings.map(reading => reading.meterNumber.trim()))
+                const meters = await find('Meter', {
+                    organization,
+                    number_in: meterNumbers,
+                    deletedAt: null,
+                })
+
+                const readingsWithValidDates = readings.filter(reading => isDateValid(reading.date))
+                const meterDatePairs = readingsWithValidDates.map(reading => ({
+                    meterNumber: reading.meterNumber.trim(),
+                    date: toISO(reading.date),
+                }))
+                const meterReadings = await MeterReading.getAll(context, {
+                    meter: { id_in: meters.map(meter => meter.id) },
```

Comment:
Seems a bit strange (or am I missing something).

If I have 500 different meters and dates, that's turn query into:
(Give me all readings, where meter is one of 500 and date is one of 500), so it can be 500 x 500 = 250_000 rows here

Reply:
Also getAll will fail for >1000 objects

Reply:
> Also getAll will fail for >1000 objects

rewrote to find

Reply:
> Seems a bit strange (or am I missing something).
> 
> If I have 500 different meters and dates, that's turn query into: (Give me all readings, where meter is one of 500 and date is one of 500), so it can be 500 x 500 = 250_000 rows here

250k readings it's very strange case: it's mean that we have 500 different dates in import file and for every 500 meters we have a reading for every of 500 dates.
I think we can set a `first` here, for example 5000. What do you think?

Reply:
> I think we can set a `first` here

If we take only the first 5000 (or any other), it means that we can skip the duplicate check for some target reading

Reply:
> 500 x 500 = 250_000 rows here

Worst thing is that in the worst case we won't be able to turn it off. And it's a normal mutation, not a deferred task

Reply:
> Should we add index here, or is it fast enough?

I don't think an index is needed here. `registerMetersReadings` is a rare request